pulp_docker/tests/functional/api/test_tagging_images.py

```diff
@@ -0,0 +1,189 @@
+# coding=utf-8
+"""Tests for tagging and untagging images."""
+import unittest
+
+from pulp_smash import api, config
+from pulp_smash.pulp3.utils import gen_repo, sync
+from pulp_smash.pulp3.constants import REPO_PATH
+
+from pulp_docker.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+from pulp_docker.tests.functional.utils import gen_docker_remote
+
+from pulp_docker.tests.functional.constants import (
+    DOCKER_TAG_PATH,
+    DOCKER_TAGGING_PATH,
+    DOCKER_REMOTE_PATH,
+    DOCKERHUB_PULP_FIXTURE_1,
+    DOCKER_UNTAGGING_PATH
+)
+from requests import HTTPError
+
+
+class TaggingTestCase(unittest.TestCase):
+    """Test case for tagging and untagging images."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide-variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+        cls.repository = cls.client.post(REPO_PATH, gen_repo())
```

Comment:
These tests operate on the resource `self.repository`, which is not cleaned up in between tests. Therefore, the tests can change the environment that they operate on. I suggest that you leave cls.repository alone (so its function is to assist with the sync only) and create a new repository and copy content into it in the a `setUp` which will run independently for each test.

Unfortunately, moving all the content around like im suggesting requires the feature that I am working on. So, for now this will do.